### PR TITLE
add postgres db for local dev and test environment

### PR DESCRIPTION
**Pull request by tomsmyers**
## What does this pull request do?

adds docker-compose service for postgres for local dev and test

## What is the intent behind these changes?

first steps to getting the hibernate stuff up and running.

### CODE REVIEW

**Review by karenmillermoj — APPROVED** [review]

File: docker-compose.yml [hunk 1]
```diff
@@ -14,5 +14,15 @@ services:
       - SPRING_PROFILES_ACTIVE=dev
       - SPRING_JPA_PROPERTIES_HIBERNATE_SHOW_SQL=false
 
+  postgres:
+    image: postgres:13
```
Comment (sldblog) [inline docker-compose.yml:18]: _suggestion_
`postgres:13-alpine` is about 50% smaller

```
$ docker images | grep postgres
postgres   13                      f51c55ac75ed        12 days ago         314MB
postgres   13-alpine               770f846cacce        2 weeks ago         159MB
```

File: build.gradle.kts [hunk 2]
```diff
@@ -8,6 +8,13 @@ configurations {
...
 dependencies {
+  // security
   implementation("org.springframework.boot:spring-boot-starter-webflux")
   implementation("org.springframework.boot:spring-boot-starter-oauth2-resource-server")
+
+  // database
+  implementation("org.springframework.boot:spring-boot-starter-data-jpa")
+  implementation("org.springframework.boot:spring-boot-starter-validation")
+  implementation("org.hibernate:hibernate-core:5.4.24.Final")
+  runtimeOnly("org.postgresql:postgresql")
```
Comment (sldblog) [inline build.gradle.kts:19]: Do we use an in-memory database for testing on CI?

File: docker-compose.yml (hunk 1 above)
Comment (tomsmyers) [reply docker-compose.yml:18]: 👍 thanks for this - i will update

File: build.gradle.kts (hunk 2 above)
Comment (tomsmyers) [reply build.gradle.kts:19]: i would prefer to not to use the in memory db at all if we can help it. 

for integration tests on circle we can run the services via docker-compose.

Comment (sldblog) [reply build.gradle.kts:19]: Totally agree -- I didn't see the PostgreSQL container spun up on the CircleCI config which is why I wondered

Comment (tomsmyers) [reply build.gradle.kts:19]: ah - yeh we don't have anything to test yet so left that out of this PR. i actually explicitly added an int test that checks the DB is down!

**Pull request merged**